src/main.rs

```diff
@@ -136,27 +152,38 @@ async fn say(ctx: &Context, msg: &Message, args: Args) -> CommandResult {
         .push(msg.channel_id)
         .push("\n")
         .build();
-    file.write_all(content_to_log.as_bytes()).expect("failed to write content to log file"); 
+    file.write_all(content_to_log.as_bytes())
+        .expect("failed to write content to log file");
     Ok(())
 }
 
-#[inline] // what does this do? 
+#[inline] // what does this do?
 async fn delete_and_send(ctx: &Context, msg: &Message, to_send: &str) -> CommandResult {
     let d = msg.delete(&ctx.http);
     let m = msg.channel_id.say(&ctx.http, &to_send);
-    tokio::try_join!(d,m)?; // do both at the same time and continue once both return Ok(). It'll quit if one returns any Err()
+    tokio::try_join!(d, m)?; // do both at the same time and continue once both return Ok(). It'll quit if one returns any Err()
     Ok(())
 }
 #[command]
 #[only_in(guilds)]
 #[aliases("s", "/s")]
-// sarcasm command for tExT lIkE tHiS. By g_w1 
+// sarcasm command for tExT lIkE tHiS. By g_w1
 async fn sarcasm(ctx: &Context, msg: &Message, args: Args) -> CommandResult {
     let mut sarcasted = sarcastify(&args.rest());
     sarcasted.insert_str(0, "@: ");
     sarcasted.insert_str(1, &msg.author.name);
-    delete_and_send(ctx, msg, &sarcasted).await?;
-    Ok(())
+    match &msg.referenced_message {
+        Some(m) => {
+            let a = m.reply(&ctx.http, &sarcasted);
+            let b = msg.delete(&ctx.http);
+            tokio::try_join!(a, b)?;
+            Ok(())
+        }
+        None => {
+            delete_and_send(ctx, msg, &sarcasted).await?;
+            Ok(())
+        }
+    }
```

Comment:
If you really want, you can just copy this diff so you dont get thet formatting.